pkg/remoteclient/fake.go

```diff
@@ -0,0 +1,142 @@
+package remoteclient
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	configv1 "github.com/openshift/api/config/v1"
+	openshiftapiv1 "github.com/openshift/api/config/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	machineapi "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	autoscalingv1 "github.com/openshift/cluster-autoscaler-operator/pkg/apis/autoscaling/v1"
+	autoscalingv1beta1 "github.com/openshift/cluster-autoscaler-operator/pkg/apis/autoscaling/v1beta1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/dynamic"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+)
+
+// fakeBuilder builds fake clients for fake clusters. Used to simulate communication with a cluster
+// that doesn't actually exist in scale testing.
+type fakeBuilder struct {
+	c              client.Client
+	cd             *hivev1.ClusterDeployment
+	controllerName hivev1.ControllerName
+	urlToUse       int
+}
+
+// Build returns a fake controller-runtime test client populated with the resources we expect to query for a
+// fake cluster.
+func (b *fakeBuilder) Build() (client.Client, error) {
+	scheme, err := machineapi.SchemeBuilder.Build()
+	if err != nil {
+		return nil, err
+	}
+
+	autoscalingv1.SchemeBuilder.AddToScheme(scheme)
+	autoscalingv1beta1.SchemeBuilder.AddToScheme(scheme)
```

Comment:
This vs
```
	scheme := runtime.NewScheme()
	hivev1.AddToScheme(scheme)
	rbacv1.AddToScheme(scheme)
```

what's the difference? the latter has no error requirement..

Reply:
Those AddToSchemes still return errors. Seems like we should always check those errors as opposed to some of these that were not. I'm going to move them to common func with the regular builder, and add error checking, otherwise probably leave them be.